squire_sdk/src/server/tournaments.rs

```diff
@@ -107,5 +107,7 @@ async fn handle_new_onlooker<S: ServerState>(state: S, id: TournamentId, mut ws:
         return;
     };
     let user = state.watch_session(session).await.unwrap();
-    state.handle_new_onlooker(id, user, ws).await;
+    if !state.handle_new_onlooker(id, user, ws).await {
+        panic!("TODO what if this fails")
+    }
```

Comment:
By the point it gets here, we've already assumed that `AnyUser::convert` returned `Ok(...)`, so it seems we're already ready to panic. Should we just panic here? Does the router handle panics nicely?

Reply:
Wait that's not right. It's not panicking in the case of not `Ok(token)`, it's just returning. So I should probably silently fail like before, then, right?